Tools/MonoGame.Effect.Compiler/Effect/EffectObject.writer.cs

```diff
@@ -142,30 +142,33 @@ public void Write(BinaryWriter writer, Options options)
             writer.Write(Header.ToCharArray());
         }
 
-        private static void WriteParameters(BinaryWriter writer, d3dx_parameter[] parameters, int count)
+        private static void WriteParameters(BinaryWriter writer, ShaderProfile profile, d3dx_parameter[] parameters, int count)
         {
             writer.Write(count);
             for (var i = 0; i < count; i++)
-                WriteParameter(writer, parameters[i]);
+                WriteParameter(writer, profile, parameters[i]);
         }
 
-        private static void WriteParameter(BinaryWriter writer, d3dx_parameter param)
+        private static void WriteParameter(BinaryWriter writer, ShaderProfile profile, d3dx_parameter param)
         {
             var class_ = ToXNAParameterClass(param.class_);
             var type = ToXNAParameterType(param.type);
+            // MojoShader encodes integers and booleans into a float.
+            var encodeAsType = (profile == ShaderProfile.OpenGL) ? EffectParameterType.Single : type;
```

Comment:
This will need to be changed for ShaderConductor. Only MojoShader encodes everything as float. ShaderConductor uses ints too. I'm pretty sure it doesn't do bools though, they will end up as ints.

Reply:
The way I see this going forward is to write a separate processor along side the current d3d/Mojo compiler.
Each Processor would write the appropriate data and the deserialized Effect should be able to handle all shaders with no assumptions about the platform or the complier. 

The Writer however is not a good place to override encodeAsType.  I will move this inside the  parameter, it will make thing much clearer.

Reply:
Here we set encodeAsType to float inside 'ConstantBufferData.mojo.cs' for integers and booleans. 
Everywhere else 'encodeAsType' equals 'type'.

https://github.com/MonoGame/MonoGame/blob/eeedfda19baaeeec82d7359f8a6c60cb13c7fce4/Tools/MonoGame.Effect.Compiler/Effect/ConstantBufferData.mojo.cs#L75-L90